oauthenticator/gitlab.py

```diff
@@ -168,36 +168,38 @@ class GitLabOAuthenticator(OAuthenticator):
             return None
 
 
-    async def _check_group_whitelist(self, user_id, access_token):
+    async def _check_group_whitelist(self, user_id, username, access_token):
         http_client = AsyncHTTPClient()
         headers = _api_headers(access_token)
         # Check if user is a member of any group in the whitelist
         for group in map(url_escape, self.gitlab_group_whitelist):
-            url = "%s/groups/%s/members/%d" % (GITLAB_API, group, user_id)
+            url = "%s/groups/%s/members/all?query=%s" % (GITLAB_API, group, username)
             req = HTTPRequest(url, method="GET", headers=headers)
             resp = await http_client.fetch(req, raise_error=False)
-            if resp.code == 200:
-                return True  # user _is_ in group
+            if resp.code == 200 and resp.body:
+                resp_json = json.loads(resp.body.decode('utf8', 'replace'))
+                for user in resp_json:
+                    if user['id'] == user_id and user['username'] == username:
```

Comment:
If is redundant to check both these, isn't it?

Reply:
Yes, I suppose checking just one should suffice as these are both values we get back from GitLab. I would pick the `id` for no particular reason, though `username` is probably also safe.

Reply:
It's fine to keep both, but the fact that both were checked made me consider if i made a faulty assumption only one check was needed. That would be the main reason for me to only use one.